File: Aliucord/src/main/java/com/aliucord/coreplugins/CorePlugins.java
package com.aliucord.coreplugins;

import android.content.Context;

import com.aliucord.coreplugins.plugindownloader.PluginDownloader;
import com.aliucord.PluginManager;
import com.aliucord.coreplugins.welcomepage.WelcomeScreen;
import com.aliucord.entities.Plugin;

import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Map.Entry;

/** CorePlugins Manager */
public final class CorePlugins {
    private static final Map<String, Plugin> corePlugins = new LinkedHashMap<>();

    /** Loads all core plugins */
    public static void loadAll(Context context) {
        corePlugins.put("CommandHandler", new CommandHandler());
        corePlugins.put("CoreCommands", new CoreCommands());
        corePlugins.put("NoTrack", new NoTrack());
        corePlugins.put("PluginDownloader", new PluginDownloader());
        corePlugins.put("SupportWarn", new SupportWarn());
        corePlugins.put("TokenLogin", new TokenLogin());
        corePlugins.put("WelcomePage", new WelcomeScreen());

        for (Entry<String, Plugin> entry : corePlugins.entrySet()) {
            Plugin p = entry.getValue();
            PluginManager.logger.info("Loading core plugin: " + entry.getKey());
            try {
                p.load(context);
            } catch (Throwable e) {
                PluginManager.logger.errorToast("Failed to load core plugin " + p.getName(), e);
            }
        }
    }

    /** Starts all core plugins */
    public static void startAll(Context context) {
        for (Entry<String, Plugin> entry : corePlugins.entrySet()) {
            Plugin p = entry.getValue();
            PluginManager.logger.info("Starting core plugin: " + entry.getKey());
            try {
                p.start(context);
            } catch (Throwable e) {
                PluginManager.logger.errorToast("Failed to start core plugin " + p.getName(), e);
            }
        }
    }
}

File: Aliucord/src/main/java/com/aliucord/coreplugins/CorePlugins.kt
package com.aliucord.coreplugins

import android.content.Context
import com.aliucord.PluginManager
import com.aliucord.coreplugins.plugindownloader.PluginDownloader
import com.aliucord.coreplugins.rn.RNAPI

/** CorePlugins Manager */
object CorePlugins {
    private var loaded = false
    private var started = false
    private val corePlugins = arrayOf(
        RNAPI(),
        Badges(),
        CommandHandler(),
        CoreCommands(),
        NoTrack(),
        PluginDownloader(),
        SupportWarn(),
        TokenLogin(),
        ButtonsAPI(),
        UploadSize(),
        DefaultStickers(),
        PrivateThreads()
    )

    /** Loads all core plugins */
    @JvmStatic
    fun loadAll(context: Context) {
        check(!loaded) { "CorePlugins already loaded" }
        loaded = true
        for (p in corePlugins) {
            PluginManager.logger.info("Loading core plugin: ${p.name}")
            try {
                p.load(context)
            } catch (e: Throwable) {
                PluginManager.logger.errorToast("Failed to load core plugin " + p.name, e)
            }
        }
    }

    /** Starts all core plugins */
    @JvmStatic
    fun startAll(context: Context) {
        check(!started) { "CorePlugins already started" }
        started = true
        for (p in corePlugins) {
            PluginManager.logger.info("Starting core plugin: ${p.name}")
            try {
                p.start(context)
            } catch (e: Throwable) {
                PluginManager.logger.errorToast("Failed to start core plugin " + p.name, e)
            }
        }
    }
}
